src/mono/mono/metadata/assembly-load-context.c

```diff
@@ -13,15 +13,30 @@
 
 GENERATE_GET_CLASS_WITH_CACHE (assembly_load_context, "System.Runtime.Loader", "AssemblyLoadContext");
 
+static GSList *alcs;
+static MonoAssemblyLoadContext *default_alc;
+static MonoCoopMutex alc_list_lock; /* Used when accessing 'alcs' */
+
+static inline void
+alcs_lock (void)
+{
+	mono_coop_mutex_lock (&alc_list_lock);
+}
+
+static inline void
+alcs_unlock (void)
+{
+	mono_coop_mutex_unlock (&alc_list_lock);
+}
+
 static void
-mono_alc_init (MonoAssemblyLoadContext *alc, MonoDomain *domain, gboolean collectible)
+mono_alc_init (MonoAssemblyLoadContext *alc, gboolean collectible)
 {
 	MonoLoadedImages *li = g_new0 (MonoLoadedImages, 1);
 	mono_loaded_images_init (li, alc);
-	alc->domain = domain;
 	alc->loaded_images = li;
 	alc->loaded_assemblies = NULL;
-	alc->memory_manager = mono_mem_manager_create_singleton (alc, domain, collectible);
+	alc->memory_manager = mono_mem_manager_create_singleton (alc, mono_get_root_domain (), collectible);
```

Comment:
On a related note, do you know if `mono_get_root_domain` is getting inlined by the compiler?

Reply:
Probably not, its not an inline function.